yt/units/yt_array.py

```diff
@@ -1370,9 +1372,25 @@ def __array_ufunc__(self, ufunc, method, *inputs, **kwargs):
             func = getattr(ufunc, method)
             if 'out' in kwargs:
                 out_orig = kwargs.pop('out')
-                out = np.asarray(out_orig[0])
+                if ufunc in multiple_output_operators:
+                    outs = []
+                    for arr in out_orig:
+                        outs.append(arr.view(np.ndarray))
+                    out = tuple(outs)
+                else:
+                    out_element = out_orig[0]
+                    if out_element.dtype.kind in ("u", "i"):
```

Comment:
```suggestion
                    if out_element.dtype.kind in "ui":
```
is equivalent but is arguably harder to read.